test/FunctionalTests/Microsoft.OData.Edm.Tests/EdmUtilTests.cs

```diff
@@ -13,6 +14,47 @@ namespace Microsoft.OData.Edm.Tests
     ///</summary>
     public class EdmUtilTests
     {
+        private static EdmEntityType customer = new EdmEntityType("NS", "Customer");
+        private static EdmEntityType city = new EdmEntityType("Sample", "City");
+        private static EdmComplexType complex = new EdmComplexType("NS.Default", "Address");
+        private static EdmEntityContainer container = new EdmEntityContainer("NS", "Default");
+        private static EdmEntitySet entitySet = new EdmEntitySet(container, "Customers", customer);
+
+        private static IEdmProperty nameProperty;
+        private static IEdmProperty addressProperty;
+        private static EdmNavigationProperty navUnderComplex;
+        private static EdmNavigationProperty navUnderCustomer;
+
+        public EdmUtilTests()
+        {
+            customer.AddKeys(customer.AddStructuralProperty("Id", EdmCoreModel.Instance.GetInt32(false)));
+            customer.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+
+            complex.AddStructuralProperty("Road", EdmCoreModel.Instance.GetString(false));
+            navUnderComplex = complex.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "City",
+                    Target = city,
+                    TargetMultiplicity = EdmMultiplicity.One,
+                });
+
+            customer.AddStructuralProperty("Address", new EdmComplexTypeReference(complex, false));
+
+            navUnderCustomer = customer.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "Cities",
+                    Target = city,
+                    TargetMultiplicity = EdmMultiplicity.Many,
+                });
+
+            container.AddElement(entitySet);

```

Comment:
This line is causing an exception to be thrown:
> System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Microsoft.OData.Edm.RegistrationHelper.AddElement[T](T element, String name, Dictionary`2 elementDictionary, Func`3 ambiguityCreator) in C:\odata.net\src\Microsoft.OData.Edm\RegistrationHelper.cs:line 57